apps/frontend/src/pages/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,152 @@
+import { ONBOARDING_FORM_DEFAULT_VALUES } from "~/pages/onboarding/libs/constants/constants.js";
+
+import { OnboardingAnswer } from "./libs/components/onboarding-answer/onboarding-answer.js";
+import {
+	type OnboardingAnswerRequestBodyDto,
+	type OnboardingFormValues,
+} from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+const Onboarding: React.FC = () => {
+	const dispatch = useAppDispatch();
+
+	const [answerIds, setAnswerIds] = useState<OnboardingAnswerRequestBodyDto>({
+		answerIds: [],
+	});
+
+	const {
+		currentQuestionIndex,
+		isLastQuestion,
+		question,
+		totalQuestionsAmount,
+	} = useAppSelector(({ onboarding }) => ({
+		currentQuestionIndex: onboarding.currentQuestionIndex,
+		isLastQuestion:
+			onboarding.currentQuestionIndex ===
+			onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+		question: onboarding.currentQuestion,
+		totalQuestionsAmount: onboarding.questions.length,
+	}));
+
+	useEffect(() => {
+		void dispatch(onboardingActions.getAll());
+	}, [dispatch]);
+
+	const { control, handleSubmit, isValid, reset } =
+		useAppForm<OnboardingFormValues>({
+			defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
+			validationSchema: oneAnswerSelectedValidationSchema,
+		});
+
+	const handleSaveAnswers = useCallback(
+		(payload: OnboardingAnswerRequestBodyDto) => {
+			// void dispatch(onboardingActions.saveAnswers(payload));
+			return payload;
+		},
+		[],
+	);
+
+	const handleNextStep = useCallback(
+		(data: OnboardingFormValues) => {
+			setAnswerIds((previousState) => {
+				const newAnswerIds = new Set([
+					...previousState.answerIds,
+					Number(data.answer),
+				]);
+
+				return {
+					answerIds: [...newAnswerIds],
+				};
+			});
+
+			if (isLastQuestion) {
+				handleSaveAnswers(answerIds);
+
+				return;
+			}
+
+			void dispatch(onboardingActions.nextQuestion());
+			reset();
+		},
+
+		[answerIds, dispatch, handleSaveAnswers, isLastQuestion, reset],
+	);
+
+	const handlePreviousStep = useCallback(() => {
+		void dispatch(onboardingActions.previousQuestion());
+	}, [dispatch]);
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			void handleSubmit(handleNextStep)(event_);
+		},
+		[handleNextStep, handleSubmit],
+	);
+
+	return (
+		<div className={styles["container"]}>
+			<div className={styles["onboarding"]}>
+				{question ? (
+					<>
+						<div className={styles["progress-bar"]}>
+							<ProgressBar
+								currentQuestionIndex={currentQuestionIndex}
+								totalQuestionsAmount={totalQuestionsAmount}
+							/>
+						</div>
+						<h2 className={styles["question"]}>{question.label}</h2>
+						<div className={styles["answers"]}>
+							<form className={styles["answers"]}>
+								{question.answers.map((answer) => (
+									<OnboardingAnswer
+										control={control}
+										key={answer.id}
+										name="answer"
+										options={[
+											{ label: answer.label, value: answer.id.toString() },
```

Comment:
Here we have only 1 option, can't we send all options to one OnboardingAnswer component?

Reply:
If I change OnboardingAnswer props  from 

```
type FormFieldProperties<T extends FieldValues> = {
	control: Control<T, null>;
	name: FieldPath<T>;
	options?: { label: string; value: string }[];
};
```

and add `answer` to props it will make these props only for this component and it can't be reusable.

Reply:
In options you should send all options, not just one. Check the implementation in #263